src/frontend/src/handler/show.rs

```diff
@@ -267,6 +268,38 @@ pub async fn handle_show_object(
                 .values(rows.into(), row_desc)
                 .into());
         }
+        ShowObject::ProcessList => {
+            let rows = {
+                let sessions_map = session.env().sessions_map();
+                sessions_map
+                    .read()
+                    .values()
+                    .map(|s| {
+                        Row::new(vec![
+                            Some(format!("{}-{}", s.id().0, s.id().1).into()),
+                            Some(s.user_name().to_owned().into()),
+                            Some(format!("{}", s.peer_addr()).into()),
+                            Some(s.database().to_owned().into()),
+                            s.elapse_since_running_sql()
+                                .map(|mills| format!("{}ms", mills).into()),
+                            s.running_sql().map(|sql| {
+                                let mut sql = sql.to_string();
+                                let old_len = sql.len();
+                                sql.truncate(1024);
+                                if old_len > sql.len() {
+                                    sql += "...";
+                                }
+                                sql.into()
```

Comment:
Maybe we can reuse `TruncatedFormatter` from @xxchan?

https://github.com/risingwavelabs/risingwave/blob/ed4a3d7bb41f861ce11737c46018a4c4cf1bec5c/src/utils/pgwire/src/pg_protocol.rs#L1073